src/webgl/primitives.js

```diff
@@ -800,59 +800,110 @@ p5.RendererGL.prototype.triangle = function(args) {
 };
 
 p5.RendererGL.prototype.ellipse = function(args) {
-  var x = args[0];
-  var y = args[1];
-  var width = args[2];
-  var height = args[3];
-  //detailX and Y are optional 6th & 7th
-  //arguments
-  var detailX = args[4] || 24;
-  var gId = 'ellipse|' + detailX;
-  if (!this.geometryInHash(gId)) {
-    var _ellipse = function() {
-      this.vertices.push(new p5.Vector(0.5, 0.5, 0));
-      this.uvs.push([0.5, 0.5]);
+  this.arc(args[0], args[1], args[2], args[3], 0 , PI * 2, 'ellipse' , args[4]);
+};
 
-      for (var i = 0; i < this.detailX; i++) {
-        var u = i / this.detailX;
-        var theta = 2 * Math.PI * u;
+p5.RendererGL.prototype.arc = function(args){
 
-        var _x = 0.5 + Math.cos(theta) / 2;
-        var _y = 0.5 + Math.sin(theta) / 2;
+   var x       = arguments[0];
+   var y       = arguments[1];
+   var width   = arguments[2];
+   var height  = arguments[3];
+   var start   = arguments[4];
+   var stop    = arguments[5];
+   var mode    = arguments[6];
+   var detailX = arguments[7] || 25;
+   var detailY = 1;
 
-        this.vertices.push(new p5.Vector(_x, _y, 0));
-        this.uvs.push(_x, _y);
+   var gId = 'arc|' + x + "|" + y + "|" + width + "|" + 
+   height + "|" + start + "|" + stop + "|" + mode + "|" + detailX + "|";
```

Comment:
as Spongman notes, you can remove any parameters here that aren't factors in the geometry and instead happen through scale/translate later (x, y, width, height)

Reply:
Fixed. Removed x, y, width and height.